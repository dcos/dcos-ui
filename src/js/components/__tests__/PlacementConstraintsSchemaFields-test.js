import PlacementValidators from "#PLUGINS/services/src/js/validators/PlacementsValidators";

const React = require("react");
const renderer = require("react-test-renderer");

const PlacementConstraintsSchemaFields = require("../PlacementConstraintsSchemaField")
  .default;

let thisValidateNoBatchErrorRestore, thisInstance;

describe("PlacementConstraintsSchemaField", () => {
  describe("with valid content", () => {
    beforeEach(() => {
      thisValidateNoBatchErrorRestore =
        PlacementValidators.validateNoBatchError;
      PlacementValidators.validateNoBatchError = () => true;
    });

    afterEach(() => {
      PlacementValidators.validateNoBatchError = thisValidateNoBatchErrorRestore;
    });

    it("displays edit constraints area", () => {
      const fieldProps = {
        formData: "[['hostname', 'MAX_PER', '1']]",
        disabled: false,
        name: "placement_constraints",
        schema: {}
      };
      thisInstance = renderer.create(
        <PlacementConstraintsSchemaFields
          fieldProps={fieldProps}
          label="label"
          schema={{}}
        />
      );

      var tree = thisInstance.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with invalid content", () => {
    beforeEach(() => {
      thisValidateNoBatchErrorRestore =
        PlacementValidators.validateNoBatchError;
      PlacementValidators.validateNoBatchError = () => false;
    });

    afterEach(() => {
      PlacementValidators.validateNoBatchError = thisValidateNoBatchErrorRestore;
    });

    it("displays form to edit constraints", () => {
      const fieldProps = {
        formData: "['hostname', 'MAX_PER', '2']",
        disabled: false,
        name: "placement_constraints"
      };

      thisInstance = renderer.create(
        <PlacementConstraintsSchemaFields
          fieldProps={fieldProps}
          label="label"
          schema={{}}
        />
      );

      var tree = thisInstance.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
