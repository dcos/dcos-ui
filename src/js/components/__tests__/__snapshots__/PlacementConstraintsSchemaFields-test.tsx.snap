// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacementConstraintsSchemaField with invalid content displays form to edit constraints 1`] = `
<div>
  <label
    className=""
  >
    label
  </label>
  <pre>
    {
  "placement_constraints": "['hostname', 'MAX_PER', '2']"
}
  </pre>
  <p
    className="form-control-feedback"
  >
    <span />
  </p>
</div>
`;

exports[`PlacementConstraintsSchemaField with valid content displays edit constraints area 1`] = `
<div
  className="pod flush-left flush-right flush-top"
>
  <h2>
    <span />
     
    
    <div
      className="tooltip-wrapper text-align-center"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span
        className="info-tooltip-icon-wrap"
      >
        <svg
          aria-label="system-circle-question icon"
          className="css-ed86y8"
          data-cy="icon"
          height={16}
          preserveAspectRatio="xMinYMin meet"
          role="img"
          viewBox="0 0 16 16"
          width={16}
        >
          <use
            xlinkHref="#system-circle-question"
          />
        </svg>
      </span>
    </div>
  </h2>
  <div
    onChange={[Function]}
  >
    <div>
      <div
        className="form-row row"
      >
        <div
          className="form-group column-12"
        >
          <a
            className="button button-primary-link button-flush"
            onClick={[Function]}
          >
            <svg
              aria-label="system-plus icon"
              className="css-e3vtaz"
              data-cy="icon"
              height={12}
              preserveAspectRatio="xMinYMin meet"
              role="img"
              viewBox="0 0 12 12"
              width={12}
            >
              <use
                xlinkHref="#system-plus"
              />
            </svg>
            <span>
              <span />
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <p
    className="form-control-feedback"
  />
</div>
`;
